packages/core/base/src/types/PaymentElement/crypto.ts

```diff
@@ -0,0 +1,25 @@
+import { CommonPaymentElementProps } from ".";
+import { CryptoPaymentMethod } from "..";
+
+export type CryptoPaymentElementProps<PM extends CryptoPaymentMethod = CryptoPaymentMethod> =
+    CommonPaymentElementProps<PM> & {
+        signer?: CryptoPaymentMethodSignerMap[PM];
+    };
+
+type CryptoPaymentMethodSignerMap = {
+    [CryptoPaymentMethod.ETH]: ETHPaymentElementSigner;
+    [CryptoPaymentMethod.SOL]: SOLPaymentElementSigner;
+};
+
+type CommonPaymentElementSignerProps = {
+    address: string;
+};
+
+// Signers
+export type ETHPaymentElementSigner = CommonPaymentElementSignerProps & {
+    signAndSendTransaction: (transaction: any) => Promise<any>;
+};
+
+export type SOLPaymentElementSigner = CommonPaymentElementSignerProps & {
+    signTransaction: (transaction: any) => Promise<any>;
+};
```

Comment:
Just to confirm, we are going to do so sign on front, broadcast on back?

Reply:
actually, to start i think its fine if we make this `signAndSendTransaction`

will change